index.js

```diff
@@ -42,7 +43,9 @@ module.exports = (function () {
           // Parent for this object is not yet in temp, adding it to pendingChildOf.
           initPush(parent, pendingChildOf, flatEl);
         }
-        delete flatEl[this.config.parent];
+        if(this.config.options.deleteParent){
+          delete flatEl[this.config.parent];
```

Comment:
I assume, there is no side-effect to this, like recursive tree builds, etc. Hope we have covered that.